src/main/java/org/sigmah/server/handler/util/Handlers.java

```diff
@@ -22,264 +22,355 @@
+						PrivacyGroupDTO groupDTO = mapper.map(p.getPrivacyGroup(), new PrivacyGroupDTO());
+					// Aggregates privacy groups among profiles.
+					if (aggretatedProfileDTO.getPrivacyGroups().get(groupDTO) != PrivacyGroupPermissionEnum.WRITE) {
+						aggretatedProfileDTO.getPrivacyGroups().put(groupDTO, p.getPermission());
+					}
+				}
+			}
+		}
+
+			List<OrgUnit> orgUnits = new ArrayList<>();
+			crawlUnits(orgUnitProfile.getOrgUnit(), orgUnits, true);
+			for (int i = 0; i < orgUnits.size(); i++) {
+				Integer orgUnitId = orgUnits.get(i).getId();
+				if (aggregatedProfiles.containsKey(orgUnitId) && orgUnitDistance.get(orgUnitId) <= i) {
+					// The OrgUnit is already inside the Map
+					// and the current profile is farther than the one inside the map so let's try another OrgUnitProfile
+					// or else
+					break;
+				}
+
+				aggregatedProfiles.put(orgUnitId, aggretatedProfileDTO);
+				orgUnitDistance.put(orgUnitId, i);
+			}
+		}
+
+		return aggregatedProfiles;
+	}
+
+	/**
+	 * Adds recursively all the OrgUnits children of a {@code user} in a collection.
+	 *
+	 * @param user
+	 *          The {@link User} from which the hierarchy is traversed.
+	 * @param units
+	 *          The current collection in which the units are added.
+	 * @param addRoot
+	 *          If the root must be added too.
+	 */
+	public static void crawlUnits(final User user, final Collection<OrgUnit> units, final boolean addRoot) {
+
+		if (user == null || user.getOrgUnitsWithProfiles() == null) {
+			// No units available.
+			return;
+		}
+
+		for (OrgUnitProfile orgUnitProfile : user.getOrgUnitsWithProfiles()) {
+			crawlUnits(orgUnitProfile.getOrgUnit(), units, addRoot);
+		}
+	}
+
+	/**
+	 * Adds recursively all the children of an unit in a collection.
+	 *
+	 * @param root
+	 *          The root unit from which the hierarchy is traversed.
+	 * @param units
+	 *          The current collection in which the units are added.
+	 * @param addRoot
+	 *          If the root must be added too.
+	 */
+	public static void crawlUnits(final OrgUnit root, final Collection<OrgUnit> units, final boolean addRoot) {
+
+		if (addRoot) {
+			units.add(root);
+		}
+
+		final Set<OrgUnit> children = root.getChildrenOrgUnits();
+		if (children != null) {
+			for (final OrgUnit child : children) {
+				crawlUnits(child, units, true);
+			}
+		}
+	}
+
+	/**
+	 * Returns if the project is visible for the given user.
+	 *
+	 * @param project
+	 *          The project.
+	 * @param user
+	 *          The user.
+	 * @return If the project is visible for the user.
+	 */
+	public static boolean isProjectVisible(final Project project, final User user) {
+		return isProjectAccessible(project, user, false);
+	}
+
+	public static boolean isProjectEditable(final Project project, final User user) {
+		return isProjectAccessible(project, user, true);
+	}
+
+	public static boolean isProjectAccessible(Project project, User user, boolean edition) {
```

Comment:
Great! I like providing two method instead of a flag argument for clarity. So, this method should be private to force call on one of the two above methods.